plugins/org.jkiss.dbeaver.tasks.native/src/org/jkiss/dbeaver/tasks/nativetool/AbstractNativeToolHandler.java

```diff
@@ -489,15 +491,26 @@ public void run() {
 
 
                 if (isLogInputStream) {
-                    String errorMessage = readStream(input.getErrorStream());
+                    CompletableFuture<Void> readInputFuture = CompletableFuture.runAsync(() -> {
```

Comment:
Just use plain thread insteda

Reply:
I would like to guarantee finishing the thread execution before printing that task is finished. Futures should be more suitable for that that plain threads
```
                logWriter.print("Task '" + task.getName() + "' finished at " + new Date() + lf);
```

Reply:
`Thread#join` it?

Reply:
Future is self describing and simpler

Reply:
I disagree